deploy/demo/quickstart/aad-pod-identity-quickstart.sh

```diff
@@ -0,0 +1,150 @@
+	else
+		echo "Azure Managed Service Identity ${1} does not exist. Creating MSI now..."
+		read MSINAME MSICLIENTID MSIRESOURCEID MSIPRINCIPALID < <(echo $(az identity create -g ${2//\"/} -n ${1//\"/} -o json | jq -r '.name, .clientId, .id, .principalId'))
+		echo "Created MSI account ${MSINAME}, and allowing account to propagate the AAD directory..."
+		sleep 30
+	fi
+}
+
+echo "Validating action of the script..."
+# Determine if demo script is to deploy demo env or clean up demo env
+# ${1//\"/} is used to remove quotes from string, which is needed for the az commands
+if [ $1 = "deploy" ]; then
+	echo "Script is to deploy AAD Pod Idenity demo."
+	validate_demo_config_file "$2"
+	validate_azure_resource_group $ASSETSRESOURCEGROUPNAME $LOCATION # Creating/Validating the Assets Resource Group
+	validate_azure_resource_group $ACCESSRESOURCEGROUPNAME $LOCATION # Creating/Validating the Access Resrouce Group
+	validate_azure_resource_group $NOACCESSRESOURCEGROUPNAME $LOCATION # Creating/Validating the No Access Resource Group
+	validate_msi_account $MSINAME $ASSETSRESOURCEGROUPNAME # Creating/Validating the Managed Service Identity
+	echo "Assigning MSI ${MSINAME} Contributor role access to access resource group ${ACCESSRESOURCEGROUPNAME}..."
+	ACCESSRESOURCEGROUPID=$(az group create -l ${LOCATION//\"/} -n ${ACCESSRESOURCEGROUPNAME//\"/} --query id -o tsv)
+	ASSIGNCONTRIBUTOR=$(az role assignment create --assignee-object-id ${MSIPRINCIPALID//\"/} --scope ${ACCESSRESOURCEGROUPID//\"/} --role Contributor 2>/dev/null)
+	echo "Completed assigning MSI ${MSINAME} Contributor role access to access resource group ${ACCESSRESOURCEGROUPNAME}."
+	echo "Assigning MSI ${MSINAME} Reader role access to access resource group ${NOACCESSRESOURCEGROUPNAME}..."
+	NOACCESSRESOURCEGROUPID=$(az group create -l ${LOCATION//\"/} -n ${NOACCESSRESOURCEGROUPNAME//\"/} --query id -o tsv)
+	ASSIGNREADER=$(az role assignment create --assignee-object-id ${MSIPRINCIPALID//\"/} --scope ${NOACCESSRESOURCEGROUPID//\"/} --role Reader 2>/dev/null)
+	echo "Completed assigning MSI ${MSINAME} Contributor role access to access resource group ${ACCESSRESOURCEGROUPNAME}."
+	echo "Creating and updating ${PODIDENTITYJSONFILENAME} from template..."
+	jq -r --arg MSINAME "$MSINAME" '.items[].metadata.name |= $MSINAME' ./aadpodidentity-template.json . > ${PODIDENTITYJSONFILENAME//\"/}
+	CLIENTID=$(jq -r --arg MSICLIENTID "$MSICLIENTID" '.items[].spec.ClientID |= $MSICLIENTID' ${PODIDENTITYJSONFILENAME//\"/} . | grep ClientID)
+	sed -i "11s/.*/$CLIENTID/g" ${PODIDENTITYJSONFILENAME//\"/}
+	RESOURCEID=$(jq -r --arg MSIRESOURCEID "$MSIRESOURCEID" '.items[].spec.ResourceID |= $MSIRESOURCEID' ${PODIDENTITYJSONFILENAME//\"/} . | grep ResourceID)
+	sed -i "12s#.*#$RESOURCEID#g" ${PODIDENTITYJSONFILENAME//\"/}
+	echo "Completed creation of ${PODIDENTITYJSONFILENAME}."
+	echo "Creating and updating ${AZUREIDENTITYBINDINGJSONFILENAME} from template..."
```

Comment:
Instead of using a template file, what do you think of using kubectl apply from stdin, which is more readable and maintainable. You can also seamlessly incorporate environment variables like so:

```bash
cat <<EOF | kubectl apply -f -
apiVersion: "aadpodidentity.k8s.io/v1"
kind: AzureIdentity
metadata:
  name: $MSINAME
spec:
  type: 0
  resourceID: $RESOURCEID
  clientID: $CLIENTID
EOF
```
[Example](https://github.com/Azure/aad-pod-identity#3-deploy-azureidentity) from README.md

Reply:
Agreed.

Reply:
I think when I was testing I used the template file as a tracking artifact for the uninstall of the demo. As long as we can output the file as an artifact to track for the uninstall that should work.